### ISSUE TITLE
Nuxt hydration mismatches with accordion

### ISSUE THREAD

**Issue by ptdev** [bootstrap-vue-next/bootstrap-vue-next#2011]
### Describe the bug

Hi,

When using the latest version of BVN with nuxt3, the Accordion component throws hydration mismatch errors, apparently due to the "id" parameter.

<img width="1013" alt="Screenshot 2024-07-10 at 11 07 23" src="https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/1280528/1970c820-91a2-436c-8612-fbf5a5e5367b">

The linked reproduction shows the hydration mismatches on the console.

Cheers.

### Reproduction

https://stackblitz.com/edit/nuxt-starter-kgt96m?file=app.vue

### Used Package Manager

npm

### Validations

- [X] Have tested with the latest version. This is still alpha version and sometime things change rapidly.
- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by VividLemon** [comment]
BAccordionItem fails because it uses `inheritAttrs: false`. https://github.com/nuxt/nuxt/issues/27910 in such cases you must supply a custom id

BAccordion fails presumably because of a bug in nuxt. The id generation in both client and server are matching the pattern of a nuxt generated id. `__BVID__nVl6Nsv1N9i-0___BV_accordion__`
(we do this by default:

```ts
  const getId = inject(idPluginKey, () => Math.random().toString().slice(2, 8))
  return `__BVID__${getId()}___BV_${suffix}__`
```
=> `'nVl6Nsv1N9i-0'.length > 6` showing that its a nuxt generated id
)

But gives me more confusion is that the nuxt module in the lib doesn't show an issue with seemingly all things equal

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/6c5f141d-2b65-43b3-be8f-94d0356686cf)

![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/b6560eeb-f974-4a84-b2f1-9de9331a1d57)

**Comment by ptdev** [comment]
Hi, 

Thanks for having a look at this.

Usually, when we use v-for to add a bunch of accordionitems, we use the index as id and that pretty much solves the issue.

But on this particular case, we're using it from inside a nuxt-content [Markdown component](https://content.nuxt.com/usage/markdown#vue-components) file so we don't really have an id unless we hard-code it on each entry, which is cumbersome. 

To solve this we tried to use useId() on the Markdown component to generate an id each time the component is called. This is the component:

```
<script setup lang="ts">
const i = useId()
</script>

<template>
    <BAccordionItem :id="i">
        <template v-slot:title>
            <ContentSlot unwrap="p" />
        </template>
        <ContentSlot :use="$slots.answer" />
    </BAccordionItem>
</template>
```

That gets called from a Markdown file like so:

```
::FaqItem
    Some question?
    
    #answer
    Some answer!
::
```

But we ran into a similar issue as on your screenshot. useId() generates a string with hyphen on the server, and with underscore on the client.

So we're getting something like this:

```
- rendered on server: id="nxh1e8XP8Ma-7"
- expected on client: id="nxh1e8XP8Ma_7"
```

Not sure if this is by design or if it's even an issue with BVN, appears to be more of a nuxt issue maybe (?)

**Comment by VividLemon** [comment]
If you're using purely `const i = useId()` with `<BAccordionItem :id="i">`, this is a nuxt issue. The id prop does nothing extra if you pass in your own id.

**Comment by ptdev** [comment]
Hi,

Yes, since we do not want to hardcode a bunch of id's we tried using useId() just to see if it would fix that problem, it doesn't.

Ideally we would just use BAccordionItem directly without having to set an id at all (as per the documentation examples), but that is apparently impossible without having hydration issues on nuxt (?)

**Comment by VividLemon** [comment]
Would have been fixed by the Vue 3.5 update
